src/hashmap.cpp

```diff
@@ -0,0 +1,68 @@
+#include "hashmap.hpp"
+#include "csv.hpp"
+#include <sparsepp/spp.h>
+
+using spp::sparse_hash_map;
+
+Hashmap::Hashmap() {};
+
+Hashmap::Hashmap(const std::string &input_filename) {
+
+    io::CSVReader<3> in(input_filename);
+    in.set_header("from", "to", "speed");
+    external_nodeid_t from; external_nodeid_t to; congestion_speed_t speed;
+    // Pre-allocate a large chunk of memory to save on all the micro-allocations
+    // that would happen if adding items one-by-one and growing as needed
+    annotations.reserve(139064548);
```

Comment:
This is the size of the wholeworld.csv file that contains all of the freeflow data currently.